plugins/modules/system_access_users.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#  Copyright: (c) 2024, Puzzle ITC, Kilian Soltermann <soltermann@puzzle.ch>
+#  GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+"""system_access_users module: Read, write, edit operations for OPNsense Users """
+
+__metaclass__ = type
+
+# https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html
+# fmt: off
+
+DOCUMENTATION = r'''
+---
+module: system_access_users
+short_description: Manage OPNsense users
+description:
+    - This module allows you to manage users on an OPNsense firewall.
+author:
+    - Kilian Soltermann (@killuuuhh)
+version_added: "1.0.0"
+options:
+    username:
+        description:
+            - The username of the OPNsense user.
+        required: true
+        type: str
+    password:
+        description:
+            - The password of the OPNsense user.
+        required: true
+        type: str
+    disabled:
+        description:
+            - Indicates whether the user account should be disabled.
+        required: false
+        default: false
+        type: bool
+    full_name:
+        description:
+            - The full name of the OPNsense user.
+        required: false
+        type: str
+    email:
+        description:
+            - The email address of the OPNsense user.
+        required: false
+        type: str
+    comment:
+        description:
+            - Additional comments or notes for the OPNsense user.
+        required: false
+        type: str
+    landing_page:
+        description:
+            - The landing page for the OPNsense user.
+        required: false
+        type: str
+    shell:
+        description:
+            - The shell for the OPNsense user.
+        required: false
```

Comment:
The shell should probably be a required param (in the opnsens eui it is not possible to create users without shell). Also the default should match that of the ui which is `/usr/sbin/nologin` whereas `/sbin/nologin` is a symlink to `/usr/sbin/nologin`

Reply:
Thank you @rekup! As in 26e1400, I decided to keep this parameter as non-required. If it's is not set, it will default to `/usr/sbin/nologin` as in the GUI.